From 7380451741be4d2950382b0886abc1714eeacb24 Mon Sep 17 00:00:00 2001
From: Sudeep Holla <sudeep.holla@arm.com>
Date: Thu, 21 Sep 2023 12:24:15 +0100
Subject: [PATCH] firmware: arm_scmi: Move power-domain driver to the pmdomain
 dir

To simplify with maintenance let's move the Arm SCMI power-domain driver
to the new pmdomain directory.

Link: https://lore.kernel.org/all/20230921113328.3208651-1-sudeep.holla@arm.com
Reviewed-by: Ulf Hansson <ulf.hansson@linaro.org>
Cc: Cristian Marussi <cristian.marussi@arm.com>
Signed-off-by: Sudeep Holla <sudeep.holla@arm.com>
[upstream commit af78e5c309c4b797168c8ce87c60ca06b26b1e73]
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I2ba574796743a8c7e99757bcf793c060f11a3056
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/476921
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 drivers/firmware/arm_scmi/Makefile                           | 1 -
 drivers/pmdomain/arm/Makefile                                | 1 +
 drivers/{firmware/arm_scmi => pmdomain/arm}/scmi_pm_domain.c | 0
 3 files changed, 1 insertion(+), 1 deletion(-)
 rename drivers/{firmware/arm_scmi => pmdomain/arm}/scmi_pm_domain.c (100%)

diff --git a/drivers/firmware/arm_scmi/Makefile b/drivers/firmware/arm_scmi/Makefile
index b31d78fa66cc..a7bc4796519c 100644
--- a/drivers/firmware/arm_scmi/Makefile
+++ b/drivers/firmware/arm_scmi/Makefile
@@ -16,7 +16,6 @@ scmi-module-objs := $(scmi-driver-y) $(scmi-protocols-y) $(scmi-transport-y)
 obj-$(CONFIG_ARM_SCMI_PROTOCOL) += scmi-core.o
 obj-$(CONFIG_ARM_SCMI_PROTOCOL) += scmi-module.o
 
-obj-$(CONFIG_ARM_SCMI_POWER_DOMAIN) += scmi_pm_domain.o
 obj-$(CONFIG_ARM_SCMI_POWER_CONTROL) += scmi_power_control.o
 
 ifeq ($(CONFIG_THUMB2_KERNEL)$(CONFIG_CC_IS_CLANG),yy)
diff --git a/drivers/pmdomain/arm/Makefile b/drivers/pmdomain/arm/Makefile
index 7128db96deac..cfcb1f6cdd90 100644
--- a/drivers/pmdomain/arm/Makefile
+++ b/drivers/pmdomain/arm/Makefile
@@ -1,3 +1,4 @@
 # SPDX-License-Identifier: GPL-2.0-only
 
 obj-$(CONFIG_ARM_SCMI_PERF_DOMAIN) += scmi_perf_domain.o
+obj-$(CONFIG_ARM_SCMI_POWER_DOMAIN) += scmi_pm_domain.o
diff --git a/drivers/firmware/arm_scmi/scmi_pm_domain.c b/drivers/pmdomain/arm/scmi_pm_domain.c
similarity index 100%
rename from drivers/firmware/arm_scmi/scmi_pm_domain.c
rename to drivers/pmdomain/arm/scmi_pm_domain.c
-- 
2.53.0

